homeassistant/components/hassio/discovery.py

```diff
@@ -0,0 +1,140 @@
+ATTR_ADDON = 'addon'
+ATTR_NAME = 'name'
+ATTR_SERVICE = 'service'
+ATTR_CONFIG = 'config'
+ATTR_COMPONENT = 'component'
+ATTR_PLATFORM = 'platform'
+ATTR_UUID = 'uuid'
+
+CONFIG_FLOW_SERVICE = ('mqtt',)
+
+
+@callback
+def async_setup_discovery(hass, hassio, config):
+    """Discovery setup."""
+    hassio_discovery = HassIODiscovery(hass, hassio, config)
+
+    # Handle exists discovery messages
+    async def async_discovery_start_handler(event):
+        """Process all exists discovery on startup."""
+        try:
+            data = await hassio.retrieve_discovery_messages()
+        except HassioAPIError as err:
+            _LOGGER.error("Can't read discover info: %s", err)
+            return
+
+        jobs = [hassio_discovery.async_process_new(discovery)
+                for discovery in data[ATTR_DISCOVERY]]
+        if jobs:
+            await asyncio.wait(jobs)
+
+    hass.bus.async_listen_once(
+        EVENT_HOMEASSISTANT_START, async_discovery_start_handler)
+
+    hass.http.register_view(hassio_discovery)
+
+
+class HassIODiscovery(HomeAssistantView):
+    """Hass.io view to handle base part."""
+
+    name = "api:hassio_push:discovery"
+    url = "/api/hassio_push/discovery/{uuid}"
+
+    def __init__(self, hass, hassio, config):
+        """Initialize WebView."""
+        self.hass = hass
+        self.hassio = hassio
+        self.config = config
+
+    async def post(self, request):
+        """Handle new discovery requests."""
+        uuid = request.match_info.get(ATTR_UUID)
+
+        # Fetch discovery data and prevent injections
+        try:
+            data = await self.hassio.get_discovery_message(uuid)
+        except HassioAPIError as err:
+            _LOGGER.error("Can't read discovey data: %s", err)
+            raise HTTPServiceUnavailable() from None
+
+        await self.async_process_new(self, data)
+        return web.Response()
+
+    async def delete(self, request):
+        """Handle remove discovery requests."""
+        data = request.json()
+
+        await self.async_process_del(self, data)
+        return web.Response()
+
+    async def async_process_new(self, data):
+        """Process add discovery entry."""
+        service = data[ATTR_SERVICE]
+        component = data[ATTR_COMPONENT]
+        platform = data[ATTR_PLATFORM]
+        config_data = data[ATTR_CONFIG]
+
+        # Read addinional Add-on info
+        try:
+            addon_info = await self.hassio.get_addon_info(data[ATTR_ADDON])
+        except HassioAPIError as err:
+            _LOGGER.error("Can't read add-on info: %s", err)
+            return
+
+        # Use config flow
+        if service in CONFIG_FLOW_SERVICE:
+            # Replace Add-on ID with name
+            data[ATTR_ADDON] = addon_info[ATTR_NAME]
+
+            await self.hass.config_entries.flow.async_init(
+                service, context={'source': 'hass.io'}, data=config_data)
+            return
+
+        # Use discovery
+        if platform is None:
```

Comment:
I don't know if we should include this functionality. All the discovery things should be phased out.

Reply:
ok, I remove it